File: Personal Practice.java/one-week-of-trouble/AirConditioner.java
public class AirConditioner {

    private boolean isOn;
    private int temperature;

    public AirConditioner() {
        isOn = false;
        temperature = 16;
    }

    public void turnOn() {
        isOn = true;
        System.out.println("AC is ON");
    }

    public void turnOff() {
        isOn = false;
        System.out.println("AC is OFF");
    }

    public void increaseTemperature() {

        if (isOn) {

            if (temperature < 30) {
                temperature++;
            }

            else {
                System.out.println("Maximum temperature reached");
            }
        }

        else {
            System.out.println("AC is OFF");
        }
    }

    public void decreaseTemperature() {

        if (isOn) {

            if (temperature > 16) {
                temperature--;
            }

            else {
                System.out.println("Minimum temperature reached");
            }
        }

        else {
            System.out.println("AC is OFF");
        }
    }

    public void displayTemperature() {
        System.out.println("Temperature: " + temperature);
    }

    public static void main(String[] args) {

        AirConditioner ac = new AirConditioner();

        ac.turnOn();

        ac.displayTemperature();

        ac.increaseTemperature();
        ac.displayTemperature();

        ac.decreaseTemperature();
        ac.displayTemperature();

        ac.turnOff();
    }
}

File: Personal Practice.java/one-week-of-trouble/AirConditioner.js
class AirConditioner {

    constructor() {
        this.isOn = false;
        this.temperature = 16;
    }

    turnOn() {
        this.isOn = true;
        console.log("AC is ON");
    }

    turnOff() {
        this.isOn = false;
        console.log("AC is OFF");
    }

    increaseTemperature() {

        if (this.isOn) {

            if (this.temperature < 30) {
                this.temperature++;
            }

            else {
                console.log("Maximum temperature reached");
            }
        }

        else {
            console.log("AC is OFF");
        }
    }

    decreaseTemperature() {

        if (this.isOn) {

            if (this.temperature > 16) {
                this.temperature--;
            }

            else {
                console.log("Minimum temperature reached");
            }
        }

        else {
            console.log("AC is OFF");
        }
    }

    displayTemperature() {
        console.log("Temperature:", this.temperature);
    }
}

let ac = new AirConditioner();

ac.turnOn();

ac.displayTemperature();

ac.increaseTemperature();
ac.displayTemperature();

ac.decreaseTemperature();
ac.displayTemperature();

ac.turnOff();